### PR TITLE
compute_ctl: Remove deprecated flag `--file-cache-on-disk`

### PR DESCRIPTION
**Pull request by sharnoff**
Builds on #5617, part of neondatabase/cloud#7516. Must not be merged until after the control plane no longer sets the flag (neondatabase/cloud#7523).

Leaving this as a draft PR until it is safe to merge.

### CODE REVIEW

**Comment by sharnoff** [comment]
neondatabase/cloud#7523 has been merged but not yet released. Once that happens, this PR will be safe to merge & release.

**Review by ololobus — APPROVED** [review]

**Comment by sharnoff** [comment]
The cloud PR this was blocked on (neondatabase/cloud#7523) was released with neondatabase/cloud#7977, reverted with neondatabase/cloud#7994, and re-released with neondatabase/cloud#8000.

So: Safe to merge this now.

**Pull request merged**
